src/test/cypress/e2e/exam/test-exam/TestExamParticipation.cy.ts

```diff
@@ -0,0 +1,119 @@
+
+                addGroupWithExercise(exam, EXERCISE_TYPE.Programming, { submission: allSuccessful, expectedScore: 100 });
+                addGroupWithExercise(exam, EXERCISE_TYPE.Programming, { submission: buildError, expectedScore: 0 });
+
+                addGroupWithExercise(exam, EXERCISE_TYPE.Quiz, { quizExerciseID: 0 });
+                addGroupWithExercise(exam, EXERCISE_TYPE.Quiz, { quizExerciseID: 0 });
+
+                addGroupWithExercise(exam, EXERCISE_TYPE.Modeling);
+                addGroupWithExercise(exam, EXERCISE_TYPE.Modeling);
+            });
+        });
+    });
+
+    it('Participates as a student in a registered test exam', () => {
+        examParticipation.startParticipation(studentOne, course, exam);
+        for (let j = 0; j < exerciseArray.length; j++) {
+            const exercise = exerciseArray[j];
+            examParticipation.openExercise(j);
+            examParticipation.makeSubmission(exercise.id, exercise.type, exercise.additionalData);
+        }
+        examParticipation.handInEarly();
+        for (let j = 0; j < exerciseArray.length; j++) {
+            const exercise = exerciseArray[j];
+            examParticipation.verifyExerciseTitleOnFinalPage(exercise.id, exercise.title);
+            if (exercise.type === EXERCISE_TYPE.Text) {
+                examParticipation.verifyTextExerciseOnFinalPage(exercise.additionalData!.textFixture!);
+            }
+        }
+        examParticipation.checkExamTitle(examTitle);
+    });
+
+    it('Checks if switching test exam tabs works', () => {
+        examParticipation.startParticipation(studentTwo, course, exam);
+        for (let j = 0; j < 20; j++) {
+            const i = between(0, exerciseArray.length - 1);
+            examParticipation.openExercise(i);
+            examParticipation.checkExerciseTitle(exerciseArray[i].id, exerciseArray[i].title);
+        }
+
+        examParticipation.handInEarly();
+    });
+
+    after(() => {
+        if (course) {
+            cy.login(admin);
+            courseManagementRequests.deleteCourse(course.id!);
+        }
+    });
+});
+
+function addGroupWithExercise(exam: Exam, exerciseType: EXERCISE_TYPE, additionalData?: AdditionalData) {
+    exerciseGroupCreation.addGroupWithExercise(exam, 'Exercise ' + generateUUID(), exerciseType, (response) => {
+        if (exerciseType == EXERCISE_TYPE.Quiz) {
+            additionalData!.quizExerciseID = response.body.quizQuestions![0].id;
+        }
+        addExerciseToArray(exerciseArray, exerciseType, response, additionalData);
+    });
+}
+
+function addExerciseToArray(exerciseArray: Array<Exercise>, type: EXERCISE_TYPE, response: any, additionalData?: AdditionalData) {
+    exerciseArray.push({ title: response.body.title, type, id: response.body.id, additionalData });
+}
+
+function between(min: number, max: number) {
+    return Math.floor(Math.random() * (max - min + 1) + min);
```

Comment:
Would It be possible to use a seed here? Otherwise a test using this function might be flaky.

Reply:
I replaced the test with a more suitable one here, so the math function is not needed anymore. Thanks for the hint!